core/common/src/TextureMapping.ts

```diff
@@ -112,6 +139,10 @@ export namespace TextureMapping { // eslint-disable-line no-redeclare
     mapMode?: TextureMapping.Mode;
     /** @internal */
     worldMapping?: boolean;
+    /** True if want to use constant LOD texture mapping for the surface texture. */
+    useConstantLod?: boolean;
+    /** Parameters for constantLod mapping mode. */
```

Comment:
You refer to constant LOD as a "mapping mode" but it's not in the `Mode` enum, instead it's controlled by a separate boolean.
What's the motivation for that vs adding it to the enum?

Reply:
I guess you still need a (non-constant) mode specified if there's a normal map, and constant LOD can only be applied to the pattern map - right?

Reply:
Might it make the API a little easier to use if - like we did for `normalMapParams` - you don't need a separate `useConstantLod` property; rather, you'd enable constant LOD by providing the `constantLodParams` property?

Reply:
I had thought about adding it to the enum, but then it would have been hard to get the texture to use it without the normal map using it unless we put a mode in the normal map data.  But then that led to the user being able to specify two different mapping modes which are not constantLod which we don't support.

Reply:
Ok. I notice you **can** use constant LOD for normal map. (I think, if no pattern map is defined, only normal map, and `useConstantLod` is `true`?)
Is there a use case for that?

Reply:
I thought the normal maps might as well be able to use constant LOD as well.  You can have all 4 combinations of normal map and texture map using constant LOD or not.  In the case of just a normal map and making it use constant LOD, I think that you would have to turn on useConstantLod in the NormalMapParams.

Reply:
As for just using the presence of the constantLodParams to specify whether to use them or not, I had thought about that as well, but though the TextureMapping.ParamProps can have unspecified properties, the TextureMapping.Params looked like everything in there was specified so I didn't want to add something to it that was optional like that.  It again made it hard to specify things for a normal map and a texture.

Reply:
I'm not too crazy about the whole thing either, but it was the best solution I could come up with after changing my mind on it a few dozen times.  I'm certainly willing to change it if we can come up with something more straight forward.